include/boost/beast/core/impl/file_stdio.ipp

```diff
@@ -105,6 +105,10 @@ open(char const* path, file_mode mode, error_code& ec)
         return;
     }
 #else
+    #if (__cplusplus <= 201402L)
+        // supported by std::fopen since c++17
+        BOOST_ASSERT(mode != file_mode::write_new);
```

Comment:
How about if we make the `file_mode::write_new` constant available only in C++17 and later? And the documentation can reflect this?